packages/lit-dev-content/site/articles/article/redux-reactive-controllers.md

```diff
@@ -0,0 +1,731 @@
+---
+title: Redux + Lit with Reactive Controllers
+publishDate: 2023-12-14
+summary: Use Reactive Controllers to integrate libraries like Redux into Lit.
+thumbnail: /images/articles/redux_lit_controller
+thumbnailExtension: webp
+tags:
+  - reactive-controllers
+  - redux
+eleventyNavigation:
+  parent: Articles
+  key: Redux + Lit with Reactive Controllers
+  order: 0
+author:
+  - elliott-marquez
+  - brian-vann
+---
+
+Lit makes it easy to create web components – reusable HTML elements with shared logic. However, different elements often have similar behaviors, and creating another element just for sharing a behavior may be excessive.
+
+[Reactive Controllers](/docs/composition/controllers/) can help with the problem of sharing logic across components without having to create a new web component. They are similar to custom hooks in React, and in this article, we will use them to integrate the state manager Redux with Lit's rendering lifecycle for a more self-contained, composable, idiomatic Lit experience.
+
+By the end of this article, you will learn how to use Reactive Controllers to integrate third party libraries into Lit by integrating Redux into Lit. To do this, we will create a Reactive Controller that selects part of a Redux state and updates a component whenever the state updates.
+
+## What is a Reactive Controller?
+
+Reactive Controllers are a programming pattern that makes it easy to share logic between components by hooking into a component’s reactive update lifecycle. They achieve this by expecting an object that exposes an interface rather than having to create a new component or subclassing like you would with a mixin.
+
+One advantage of the Reactive Controller pattern is that it creates a **with** relationship rather than an **is** relationship. For example, a component that uses a Reactive Controller that incorporates Redux logic is a **component with Redux selector abilities**, whereas a mixin that does the same would mean that the **component is a Redux selector component**. This type of composability results in code that is more portable, self-contained, and easier to refactor. This is because components that inherit via subclassing are more closely coupled with the logic they inherit.
```

Comment:
Is it worth mentioning for readers familiar with React that Reactive Controllers are like hooks whereas mixins are like higher order components? It also allows parallels below how the `SelectorController` is much like the `useSelector` hook from `react-redux`.